core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/io/ImportRequest.java

```diff
@@ -0,0 +1,143 @@
+	
+	@NotNull
+	private Attachment attachment;
+	
+	public ImportRequest() {
+		super(DatastoreLockContextDescriptions.IMPORT);
+	}
+	
+	public ImportRequest(String lockContext, String parentLockContext) {
+		super(lockContext, parentLockContext);
+	}
+	
+	final void setAttachment(Attachment attachment) {
+		this.attachment = attachment;
+	}
+	
+	protected Attachment attachment() {
+		return attachment;
+	}
+	
+	@Override
+	public final ImportResponse doExecute(TransactionContext context) {
+		context.log().info("Importing components from source file '%s'.", this.attachment.getFileName());
+		try {
+			InternalAttachmentRegistry iar = (InternalAttachmentRegistry) context.service(AttachmentRegistry.class);
+			File attachment = iar.getAttachment(this.attachment.getAttachmentId());
+			
+			ImportDefectAcceptor defectsAcceptor = new ImportDefectAcceptor(this.attachment.getFileName());
+			doValidate(context, attachment, defectsAcceptor, context.service(IProgressMonitor.class));
+			
+			final List<ImportDefect> defects = defectsAcceptor.getDefects();
+			if (!CompareUtils.isEmpty(defects)) {
+				return ImportResponse.defects(defects);
+			} else {
+				final Set<ComponentURI> visitedComponents = Sets.newHashSet();
+				doImport(context, attachment, visitedComponents::add, context.service(IProgressMonitor.class));
+				return ImportResponse.success(visitedComponents);
+			}
+			
+		} catch (ApiException e) {
+			throw e;
+		} catch (Exception e) {
+			String error = "Unexpected error happened during the import of the source file: " + attachment.getFileName();
+			context.log().error(error, e);
+			return ImportResponse.error(error);
+		} finally {
+			context.log().info("Finished importing components from source file '%s'.", this.attachment.getFileName());
+		}
+	}
+
+//	/**
+//	 * Logs the import activity with the given message.
+//	 * 
+//	 * @param message the message to be logged.
+//	 */
+//	protected void logImportActivity(final String message) {
+//		LogUtils.logImportActivity(context.log(), context.author(), context.path(), message);
+//	}
+//	
+//	/**
+//	 * Logs the import warning with the given message.
+//	 * 
+//	 * @param message the message to be logged.
+//	 */
+//	protected void logImportWarning(final String message) {
+//		LogUtils.logImportWarning(context.log(), context.author(), context.path(), message);
+//	}
+	
+	/**
+	 * Subclasses optionally provider validation functionality to verify the integrity of the attachment before proceeding to the actual import in {@link #doImport(File, IProgressMonitor)}.
+	 * @param context - the context to run the validation on
+	 * @param attachment - the file attachment to validate
+	 * @param defectsAcceptor - the acceptor that collects {@link ImportDefect}s through a few helper methods
+	 * @param monitor - the monitor that can be used to track progress
+	 * @return a {@link List} of {@link ImportDefect} instances that can aid the user pinpoint and resolve the errors/issues
```

Comment:
There is no return value, the import defect collecting object consumes all messages instead.